protocol/message_persistence.go

```diff
@@ -1421,77 +1432,127 @@ func (db sqlitePersistence) SaveMessages(messages []*common.Message) (err error)
 	return
 }
 
-func (db sqlitePersistence) SavePinMessages(messages []*common.PinMessage) (err error) {
+type insertPinMessagesQueries struct {
+	selectStmt  string
+	insertStmt  *sql.Stmt
+	updateStmt  *sql.Stmt
+	transaction *sql.Tx
+}
```

Comment:
This is a nice approach.

Reply:
I remember I once talked to @cammellos about the usage of a query builder library (not ORMs :rage1:). For example http://doug-martin.github.io/goqu/ Nothing magical, just type safety, and a unified way to compose queries.

Right now, the problem I see is that concatenating strings is a suboptimal abstraction that leads to various ad-hoc solutions and certainly a lot of duplication (since it's hard to compose strings from different functions).

What are your thoughts @Samyoul? Would you like to see an experiment with such a library? Have you tried any library in this space?

Reply:
Fundamentally @ilmotta I agree with you. One of the first things I wanted to do with this repo was overhaul the database interactions because there were/are so many persistence packages, mostly written with pure SQL. There are already a number of places in this repo where we've implemented various ad-hoc query builder solutions.

Settings is one stand out example, and may even be a good candidate for an experiment with a query building library.

I'd be interest in how well joining, grouping, having and more complex queries could be handled, looking over the docs of `goqu` it seems that it has robust support for quite complex queries.

So another candidate could be user identity, as there is a single row for user data and multiple rows for the user images.

Reply:
Hey @Samyoul, sorry I couldn't reply because I didn't work last Friday :tada: 

It's great to hear you're interested in seeing where this experiment could go. There's enough quorum! I'll try to cook something after I finish the feature I'm working on.

> I'd be interest in how well joining, grouping, having and more complex queries could be handled, looking over the docs of goqu it seems that it has robust support for quite complex queries.

I'm curious too to understand where `goqu` "breaks" so to speak, because that would help us understand where it can be useful and where we should avoid it (if there is any such scenario).

> So another candidate could be user identity, as there is a single row for user data and multiple rows for the user images.

Sounds reasonable! And we can do the improvements piecemeal since we can always convert the query back to a string.